lib/internal/net.js

```diff
@@ -9,3 +9,9 @@ function isLegalPort(port) {
     return false;
   return +port === (port >>> 0) && port >= 0 && port <= 0xFFFF;
 }
+
+
+function assertPort(port) {
+  if (typeof port !== 'undefined' && !isLegalPort(port))
```

Comment:
@trevnorris: we do have a test that uses `undefined` (https://github.com/dirceu/node/blob/fix-5727/test/parallel/test-net-listen-port-option.js#L7). Should I change it to explicitly check if the port is a number?

@jasnell: got it. Speaking of which, wouldn't be clearer to use something like this?

```js
    let port = h.port | 0;
    assertPort(port);
    if (h.host)
      listenAfterLookup(port, h.host, backlog, h.exclusive);
    else
      listen(self, null, port, 4, backlog, undefined, h.exclusive);
```

Otherwise, is there anything else that could be improved on the PR?